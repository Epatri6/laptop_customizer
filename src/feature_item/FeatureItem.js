import React from 'react';
// Normalizes string as a slug - a string that is safe to use
// in both URLs and html attributes
import slugify from 'slugify';
import USCurrencyFormat from '../util/USCurrencyFormat';
import './FeatureItem.css';

export default class FeatureItem extends React.Component {

  render() {
    const {id, item, featureName, selectedFeature, updateFeature} = this.props;
    return (
      <div key={id} className="feature__item">
        <input
          type="radio"
          id={id}
          className="feature__option"
          name={slugify(featureName)}
          checked={item.name === selectedFeature.name}
          onChange={(e) => updateFeature(featureName, item)}
        />
        <label htmlFor={id} className="feature__label">
          {item.name} ({USCurrencyFormat.format(item.cost)})
        </label>
      </div>
    );
  }

  static defaultProps = {
    id: '',
    item: {},
    featureName: '',
    selectedFeature: {},
    updateFeature: () => {}
  }

}