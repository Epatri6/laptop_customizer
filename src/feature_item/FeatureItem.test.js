import React from 'react';
import ReactDOM from 'react-dom';
import FeatureItem from './FeatureItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Testing ShoppingCart Component', () => {

  const testProps = {
    id: '1',
    item:
      {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
    featureName: 'Processor',
    selectedFeature: 
      {
        name: 'Professor X AMD Fire Breather with sidewinder technology',
        cost: 1200
      },
    updateFeature: () => {},
    selectedFeatureAlt: 
      {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      }
  };

  const {id, item, featureName, selectedFeature, updateFeature, selectedFeatureAlt} = testProps;

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FeatureItem id={id} item={item} featureName={featureName} selectedFeature={selectedFeature} updateFeature={updateFeature} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FeatureItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const wrapper = shallow(<FeatureItem id={id} item={item} featureName={featureName} 
      selectedFeature={selectedFeature} updateFeature={updateFeature} />);
    expect(toJson(wrapper)).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const wrapper = shallow(<FeatureItem />);
    expect(toJson(wrapper)).toMatchSnapshot();  
  });

  it('renders as expected when item is selected', () => {
    const wrapper = shallow(<FeatureItem id={id} item={item} featureName={featureName} 
      selectedFeature={selectedFeatureAlt} updateFeature={updateFeature} />);
    expect(toJson(wrapper)).toMatchSnapshot();  
    });
});