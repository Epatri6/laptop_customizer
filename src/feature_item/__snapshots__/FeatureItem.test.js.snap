// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingCart Component renders as expected 1`] = `
<div
  className="feature__item"
  key="1"
>
  <input
    checked={false}
    className="feature__option"
    id="1"
    name="Processor"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor="1"
  >
    17th Generation Intel Core HB (7 Core with donut spare)
     (
    $700.00
    )
  </label>
</div>
`;

exports[`Testing ShoppingCart Component renders as expected when item is selected 1`] = `
<div
  className="feature__item"
  key="1"
>
  <input
    checked={true}
    className="feature__option"
    id="1"
    name="Processor"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor="1"
  >
    17th Generation Intel Core HB (7 Core with donut spare)
     (
    $700.00
    )
  </label>
</div>
`;

exports[`Testing ShoppingCart Component renders as expected without props 1`] = `
<div
  className="feature__item"
  key=""
>
  <input
    checked={true}
    className="feature__option"
    id=""
    name=""
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor=""
  >
     (
    $NaN
    )
  </label>
</div>
`;
